### PR TITLE
small refines

### PR DESCRIPTION
**Pull request by XenoAmess**

### CODE REVIEW

**Comment by Spasi** [comment]
Hey @XenoAmess,

Could you please squash the 7 commits and update the PR? Also, could you explain what problem these changes are solving?

**Comment by XenoAmess** [comment]
> Hey @XenoAmess,
> 
> Could you please squash the 7 commits and update the PR? Also, could you explain what problem these changes are solving?

squash done.
explaining about this commit is added in the new commit.

File: modules/generator/src/main/java/org/lwjgl/generator/util/ModuleInfoGen.java [hunk 1]
```diff
@@ -67,15 +67,15 @@ public static void main(String[] args) throws IOException {
         }
 
         @Override public boolean equals(Object other) {
-            return name.equals(((Module)other).name);
+            return other != null && (other instanceof Module) && Objects.equals(name, ((Module)other).name);
```
Comment (kenzierocks) [inline modules/generator/src/main/java/org/lwjgl/generator/util/ModuleInfoGen.java]: `other != null` is redundant with `other instanceof Module`, which already returns `false` when `other` is `null`.

**Comment by httpdigest** [comment]
I think Spasi's question was more in the sense of "did you actually experience an issue/bug when running the code and your changes actually fix such observed bug" and not so much of a theoretical "it looks like we should do a null-pointer check here" motivation.

**Comment by XenoAmess** [comment]
> I think Spasi's question was more in the sense of "did you actually experience an issue/bug when running the code and your changes actually fix such observed bug" and not so much of a theoretical "it looks like we should do a null-pointer check here" motivation.

Haven't meet any bugs about it.
I just see it when learing about the codes.

File: modules/generator/src/main/java/org/lwjgl/generator/util/ModuleInfoGen.java (hunk 1 above)
Comment (XenoAmess) [reply modules/generator/src/main/java/org/lwjgl/generator/util/ModuleInfoGen.java]: exactly.

**Pull request closed without merging**